lib/opendax/vault.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+module Opendax
+  class Vault
+    def vault_secrets_path
+      'config/vault-secrets.yml'
+    end
+
+    def exec(command)
+      `docker-compose exec vault sh -c '#{command}'`
+    end
+
+    def unseal(keys)
+      keys.each { |key| exec("vault operator unseal #{key}") }
+    end
+
+    def login(root_token)
+      exec("vault login #{root_token}")
+    end
+
+    def secrets(command, endpoints, options = '')
+      endpoints.each { |endpoint| exec("vault secrets #{command} #{options} #{endpoint}") }
+    end
+
+    def save_output(vault_init_output)
+      File.write(vault_secrets_path, YAML.dump(vault_init_output))
+    end
+
+    def init
+      YAML.safe_load(exec('vault operator init -format yaml --recovery-shares=3 --recovery-threshold=2'))
+    end
+
+    def setup
+      puts '----- Initializing the Vault -----'
+      vault_init_output = init
```

Comment:
Would be nice to add a status check and halt initialization if Vault is already initialized